src/Disks/IO/ThreadPoolRemoteFSReader.cpp

```diff
@@ -66,14 +67,32 @@ ThreadPoolRemoteFSReader::ThreadPoolRemoteFSReader(size_t pool_size, size_t queu
 
 std::future<IAsynchronousReader::Result> ThreadPoolRemoteFSReader::submit(Request request)
 {
+    auto * fd = assert_cast<RemoteFSFileDescriptor *>(request.descriptor.get());
```

Comment:
There is increment of `ThreadpoolReaderSubmit` at the end of this method, lets move it to the top to include  `reader.set` and `reader.seek` as well? Or add a separate counter for them? Seems it should be useful to measure them as well (just in case someone makes `seek` not as cheap as it is now in `ReadBufferFromRemoteFSGather`).